packages/relay-compiler/generic/core/RelayWatchmanClient.js

```diff
@@ -17,6 +17,15 @@ const watchman = require('fb-watchman');
 class RelayWatchmanClient {
   _client: any;
 
+  static createIfAvailable(): Promise<?RelayWatchmanClient> {
+    return new Promise((resolve) => {
+      const client = new RelayWatchmanClient();
+      client.on('error', () => resolve(null));
```

Comment:
I based it on the [node watchman docs here](https://facebook.github.io/watchman/docs/nodejs.html#checking-for-watchman-availability) where they check capabilities for `relative_root`. 

Invoking `client.hasCapability` when watchman is not installed will cause the `on('error')` callback to be invoked (or a `Unhandled 'error' event` to be thrown if not implemented) and never execute the function in the `then` callback. I'm not able to catch this error by chaining a `.catch` or rewriting as an async function and wrapping in a `try/catch`. I'm investigating why that's the case now. 

I could check the error in `on('error')` to make sure it's `ENOENT` before resolving `null`.